src/bp-messages/classes/class-bp-messages-thread.php

```diff
@@ -419,12 +444,63 @@ public static function get_messages( $thread_id = 0, $args = array() ) {
 		return apply_filters( 'bp_messages_thread_get_messages', (array) $messages, (int) $thread_id, (array) $r );
 	}
 
+	/**
+	 * Get latest thread message.
+	 *
+	 * @since 11.0.0
+	 *
+	 * @global BuddyPress $bp The one true BuddyPress instance.
+	 * @global wpdb $wpdb WordPress database object.
+	 *
+	 * @param int $thread_id The message thread ID.
+	 * @return object|null
+	 */
+	public static function get_latest_thread_message( $thread_id ) {
+		global $wpdb;
+
+		$message = wp_cache_get( $thread_id, 'bp_messages_thread_latest_message' );
```

Comment:
This new cache group should probably be setup into this function:
https://github.com/buddypress/buddypress/blob/master/src/bp-messages/classes/class-bp-messages-component.php#L417|L428

I believe this was the initial purpose of: https://buddypress.trac.wordpress.org/ticket/8590